store/store_test.go

```diff
@@ -268,6 +269,29 @@ func TestEDSStore(t *testing.T) {
 		})
 	}
 
+	t.Run("HasQ4", func(t *testing.T) {
+		dir := t.TempDir()
+		edsStore, err := NewStore(paramsNoCache(), dir)
+		require.NoError(t, err)
+
+		square, roots := randomEDS(t)
+		randHeight := uint64(8)
+
+		err = edsStore.PutODSQ4(ctx, roots, randHeight, square)
+		require.NoError(t, err)
+
+		has, err := edsStore.HasQ4ByHash(ctx, roots.Hash())
```

Comment:
Nit: theoretically before `PutODSQ4` we want to check that file do not exist.

Reply:
done https://github.com/celestiaorg/celestia-node/pull/4035/commits/fb61e8dfa4e8a10c1baaf4df8dd3aa4df9f6bb48